File: mobile/ios/SpeakeasyCrypto/Vault/VaultModels.swift
import Foundation

// Models for Vault Items

struct VaultItem: Codable {
    var id: UUID
    var type: VaultItemType
    var createdAt: Date
    var updatedAt: Date
    var wrappedFileKeyB64: String?
    var nonceB64: String
    var ciphertextPath: String?     // Local file path
    var attachmentId: UUID?         // If uploaded to server
    var mimeType: String?
    var ciphertextSizeBytes: Int64?
    var displayNameEnc: String?     // Encrypted display name
}

enum VaultItemType: String, Codable {
    case note
    case file
    case photo
    case video
    case attachmentRef
}

struct VaultAttachmentRef: Codable {
    var attachmentId: UUID
    var storageKey: String
    var sha256CiphertextB64: String
    var encAlg: String
    var nonceB64: String
    var sizeBytes: Int64
    var mimeType: String?
}

File: mobile/android/crypto/vault/VaultModels.kt
package com.speakeasy.crypto.vault

import java.util.UUID
import java.util.Date

// Models for Vault Items

data class VaultItem(
    val id: UUID,
    val type: VaultItemType,
    val createdAt: Date,
    val updatedAt: Date,
    val wrappedFileKeyB64: String?,
    val nonceB64: String,
    val ciphertextPath: String?,    // Local file path
    val attachmentId: UUID?,        // If uploaded to server
    val mimeType: String?,
    val ciphertextSizeBytes: Long?,
    val displayNameEnc: String?     // Encrypted display name
)

enum class VaultItemType {
    NOTE,
    FILE,
    PHOTO,
    VIDEO,
    ATTACHMENT_REF
}

data class VaultAttachmentRef(
    val attachmentId: UUID,
    val storageKey: String,
    val sha256CiphertextB64: String,
    val encAlg: String,
    val nonceB64: String,
    val sizeBytes: Long,
    val mimeType: String?
)
